Mac-App/Mac-App/Transformers/NavigationTransformer.swift

```diff
@@ -32,7 +32,7 @@ final class NavigationTransformer: StateTransforming, StateRepresentableViewInpu
     }
 
     func startListening() {
-        cancellable = stateObserver.$currentState.sink { [unowned self] in self.emitNewState($0) }
+        cancellable = stateObserver.$currentState.receive(on: DispatchQueue.main).sink { [unowned self] in self.emitNewState($0) }
```

Comment:
This transformer takes care of views transition, thus we need to execute state changes in the main thread.